Tests/IntegrationTests/PostgresClientTests.swift

```diff
@@ -0,0 +1,54 @@
+@_spi(ConnectionPool) import PostgresNIO
+import XCTest
+import NIOPosix
+import Logging
+import Atomics
+
+@available(macOS 13.0, iOS 16.0, tvOS 16.0, watchOS 9.0, *)
+final class PostgresClientTests: XCTestCase {
+
+    func testGetConnection() async throws {
+        var mlogger = Logger(label: "test")
+        mlogger.logLevel = .debug
+        let logger = mlogger
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 8)
```

Comment:
Would `System.coreCount` be more representative here?

Reply:
This is mainly to ensure that concurrent access is really, really safe.